Src/ILGPU/Runtime/AcceleratorStream.cs

```diff
@@ -69,6 +69,12 @@ protected AcceleratorStream(Accelerator accelerator)
         /// <returns>A scoped binding object.</returns>
         public ScopedAcceleratorBinding BindScoped() => Accelerator.BindScoped();
 
+        /// <summary>
+        /// Adds a profiling marker into the stream.
+        /// </summary>
+        /// <returns>The profiling marker.</returns>
+        public abstract ProfilingMarker AddProfilingMarker();
```

Comment:
Question: Should we also provide an extension/instance method similar to `Synchronize` that adds a marker to the default stream? However, this might be a bad idea as this functionality will only apply to the default stream. @MoFtZ what do you think?

Reply:
It probably makes sense to add.